clientconn.go

```diff
@@ -1394,6 +1403,12 @@ func (ac *addrConn) getState() connectivity.State {
 	return ac.state
 }
 
+func (ac *addrConn) getCurAddr() resolver.Address {
+	ac.mu.Lock()
+	defer ac.mu.Unlock()
```

Comment:
Seems like this will be called on the hot path. How about we don't use defer, if that's the case?